homeassistant/components/incharge/sensor.py

```diff
@@ -0,0 +1,136 @@
+"""Get status for all stations for InCharge."""
+from __future__ import annotations
+
+import datetime
+import logging
+
+import async_timeout
+import requests
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+SENSOR_TYPES: dict[str, SensorEntityDescription] = {
+    "total_energy_consumption": SensorEntityDescription(
+        key="total_energy_consumption",
+        name="Total Energy Consumption",
+        suggested_display_precision=2,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    )
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the InCharge sensor."""
+    # Add entities for each charging station
+    incharge_data_coordinator = hass.data.setdefault(DOMAIN, {})[config_entry.entry_id]
+    response = await hass.async_add_executor_job(
+        incharge_data_coordinator.incharge_api.get_stations
+    )
+    for station in response.json()["stations"]:
```

Comment:
Why don't you have your api just return the dictionary instead of the response?

Reply:
The auth endpoint returns the jwt token via the header. Not sure if there are other endpoints returning information via headers. I simplified the package by just passing through the full response object rather than having a specific implementation for each endpoint. What if there are multiple endpoints later on passing through data via headers? Any thoughts on this approach?